data_science/01.00-addition-laws.py

```diff
@@ -0,0 +1,416 @@
+
+    This is a simple addition because each number can only come up once!
+    """)
+
+    mo.hstack([plt.gcf(), die_result])
+    return die_prob, die_result, selected
+
+
+@app.cell(hide_code=True)
+def _(mo):
+    mo.md(
+        """
+        ## 🚫 Mutually Exclusive Events
+
+        Events are mutually exclusive when they cannot happen at the same time.
+
+        For mutually exclusive events A and B:
+
+        $P(A \cup B) = P(A) + P(B)$
+
+        Let's visualize this with a weather example!
+        """
+    )
+    return
+
+
+@app.cell
+def _(mo, rainy_prob, sunny_prob):
+    mo.hstack([sunny_prob, rainy_prob])
+    return
+
+
+@app.cell
+def _(mo):
+    # Weather probability selector
+    sunny_prob = mo.ui.slider(
+        value=0.3,
+        start=0,
+        stop=1,
+        step=0.1,
+        label="P(Sunny)"
+    )
+    rainy_prob = mo.ui.slider(
+        value=0.4,
+        start=0,
+        stop=1,
+        step=0.1,
+        label="P(Rainy)"
+    )
+    return rainy_prob, sunny_prob
+
+
+@app.cell
+def _(mo, plt, rainy_prob, sunny_prob, venn2):
+    weather_total = sunny_prob.value + rainy_prob.value
+
+    # Create Venn diagram
+    plt.figure(figsize=(8, 5))
+    v = venn2(subsets=(sunny_prob.value, rainy_prob.value, 0),
+              set_labels=('Sunny', 'Rainy'))
+
+    # Customize colors
+    if v:
+        v.get_patch_by_id('10').set_color('#f1c40f')
+        v.get_patch_by_id('01').set_color('#3498db')
+
+    plt.title('Mutually Exclusive Weather Events')
+
+    weather_result = mo.md(f"""
+    ### Weather Probability Analysis
+
+    P(Sunny) = {sunny_prob.value}
+    P(Rainy) = {rainy_prob.value}
+    P(Sunny or Rainy) = {weather_total}
+
+    Status: {"✅ Valid" if weather_total <= 1 else "❌ Invalid - Total exceeds 1!"}
+    """)
+
+    mo.hstack([plt.gcf(), weather_result])
+    return v, weather_result, weather_total
+
+
+@app.cell(hide_code=True)
+def _(mo):
+    mo.md(
+        """
+        ## 🤝 General Addition Rule
+
+        When events can overlap, we need to subtract the overlap:
+
+        $P(A \cup B) = P(A) + P(B) - P(A \cap B)$
+
+        Let's explore this with a student club example!
+        """
+    )
+    return
+
+
+@app.cell
+def _(math_prob, mo, overlap_prob, science_prob):
+    mo.hstack([math_prob, science_prob, overlap_prob])
+    return
+
+
+@app.cell
+def _(mo):
+    # Student club membership
+    math_prob = mo.ui.slider(
+        value=0.4,
+        start=0,
+        stop=1,
+        step=0.1,
+        label="P(Math Club)"
+    )
+    science_prob = mo.ui.slider(
+        value=0.3,
+        start=0,
+        stop=1,
+        step=0.1,
+        label="P(Science Club)"
+    )
+    overlap_prob = mo.ui.slider(
+        value=0.1,
+        start=0,
+        stop=0.3,
+        step=0.1,
+        label="P(Both Clubs)"
+    )
+    return math_prob, overlap_prob, science_prob
+
+
+@app.cell
+def _(math_prob, mo, overlap_prob, plt, science_prob, venn2):
+    # Calculate total probability
```

Comment:
A venn diagram is a nice way to teach this concept. I noticed the float formatting for venn diagrams can become off.